File: Projects/Project-Euler/P22.js
() => {
  const
  fs = require('fs'),    
  buffer = fs.readFileSync(process.cwd() + '/public/Projects/Project-Euler/P22.txt'),
  names = String(buffer).split(',').sort(),
  scores = {};

  for (let i = 0; i <= 90 - 65; i++) {
    scores[String.fromCharCode(65 + i)] = i + 1;
  }

  return names.reduce((total, name, index) => {
    return total + (index + 1) * name.slice(1, -1).split('').reduce((score, letter) => {
      return score + scores[letter];
    }, 0);
  }, 0);
}

File: Projects/Project-Euler/P22.py
with open('/public/Projects/Project-Euler/P22.txt', 'r') as f:
  data = f.read().replace('"', '').split(',')
  data.sort()

ascii_offset = 64
scores = {}
result = 0

for i in range(1, 27):
  scores[chr(i + ascii_offset)] = i

for i in range(0, len(data)):
  score = 0

  for char in list(data[i]):
    score += scores.get(char)
  
  result += score * (i + 1)

print(result)